tests/unit/parsec/test_fileparse.py

```diff
@@ -357,6 +358,31 @@ def test_read_and_proc_jinja2():
                           viewcfg=viewcfg)
         assert r == ['a=Cylc']
 
+def test_read_and_proc_cwd():
+    """The template processor should be able to read workflow files.
+
+    This relies on moving to the config dir during file parsing. 
+    """
+    with tempfile.TemporaryDirectory() as td:
+        sdir = Path(td + "/sub")
+        sdir.mkdir()
+        for sub in ["a", "b", "c"]:
+            (sdir / Path(sub)).touch()
+        viewcfg = {
+            'empy': False, 'jinja2': True,
+            'contin': False, 'inline': False
+        }
+        with open(td / Path("a.conf"), 'w') as tf:
+            tf.write(
+"""#!Jinja2
+{% from "os" import listdir %}
+{% for f in listdir("sub") %}
+{{f}}
+{% endfor %}
+""")
+            tf.flush()
+            r = read_and_proc(fpath=tf.name, viewcfg=viewcfg)
+            assert sorted(r) == ['a', 'b', 'c']
```

Comment:
Possible alternative

- Slightly faster
- Less code
- Uses builtin `test_path` fixture which clears files up when the test passes.
- Use implied type forcing of Path's `/` method.

```suggestion

def test_read_and_proc_cwd(tmp_path):
    """The template processor should be able to read workflow files.

    This relies on moving to the config dir during file parsing.
    """
    sdir = tmp_path / "sub"
    sdir.mkdir()
    [(sdir / s).touch() for s in ["a", "b", "c"]]
    viewcfg = {
        'empy': False, 'jinja2': True,
        'contin': False, 'inline': False
    }
    tf = tmp_path / "a.conf"
    tf.write_text(
        '#!Jinja2\n'
        '{% from "os" import listdir %}\n'
        '{% for f in listdir("sub") %}\n'
        '{{f}}\n'
        '{% endfor %}'
    )
    with open(tf, 'r') as fh:
        r = read_and_proc(fpath=fh.name, viewcfg=viewcfg)
    assert sorted(r) == ['a', 'b', 'c']

```

Reply:
FYI, this may be considered an anti-pattern, I think some linters will flag it.

```diff
-       for sub in ["a", "b", "c"]:
-           (sdir / Path(sub)).touch()
+      [(sdir / s).touch() for s in ["a", "b", "c"]]
```

The reason being that list-comprehensions are not expected to have side-effects (touching files in this case).

Reply:
That seems reasonable.

Reply:
It also creates a list `[None, None, None]` for no reason.

Reply:
@wxtim - is there anything else in your suggestions above that you particularly think we should keep?

Note I wrapped it all in the `tempfile.TemporaryDirectory()` context manager (which you removed) for the automatic cleanup.

Reply:
Pytest does automatic cleanup on tmp_path, which is why that's the thing I'd keep out of that lot. Also, if you need to debug this later you know the files will be in `/var/tmp/pytest-of-${USER}/pytest-current`

Reply:
Done.